### ISSUE TITLE
Expense - Submit button appears for archived workspace chat if delayed submission is enabled

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#49169]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.33-4
**Reproducible in staging?:** Y
**Reproducible in production?:**Y
**Email or phone of affected tester (no customers):** nathanmulugetatesting+1469@gmail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Navigate to staging.new.expensify.com
2. Sign up with a new Gmail account
3. Create two workspaces
4. Invite a member on the second workspace
5. Enable workflows
6. Enable delayed submission and make the frequency "Manually"
7. Create an expense as the employee
8. Delete the workspace as the admin

## Expected Result:
Submit button doesn't show on the archived workspace chat on the expense

## Actual Result:
Submit button shows on both the employee's and admin's archived workspace chat and clicking on the button results in an error

## Workaround:
Unknown

## Platforms:

- [x] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/bca2d8f4-6b61-49d7-a214-ae98828881ed

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by IuliiaHerets** [comment]
@johncschuster  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by etCoderDysto** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Expense - Submit button appears for archived workspace chat if delayed submission is enabled 
### What is the root cause of that problem?
We display when report is archived 
here 
https://github.com/Expensify/App/blob/00bc167940607f1418109f00e2b4e2d7d4ae858b/src/components/MoneyReportHeader.tsx#L126
And here
https://github.com/Expensify/App/blob/00bc167940607f1418109f00e2b4e2d7d4ae858b/src/components/ReportActionItem/ReportPreview.tsx#L187
### What changes do you think we should make in order to solve the problem?
We should hide the button when report is archived 
here
```
    const shouldShowSubmitButton = isDraft && reimbursableSpend !== 0 && !hasAllPendingRTERViolations && !isArchivedReport;

```
And here
```
const shouldShowSubmitButton = isOpenExpenseReport && reimbursableSpend !== 0 && !showRTERViolationMessage && !isArchivedReport;
```
### What alternative solutions did you explore? (Optional)

**Comment by etCoderDysto** [comment]
> [!NOTE] 
> *A kind reminder for C+:*

The fist section of the first proposal's permalink edit that points to [shouldShowSubmitButton](https://github.com/Expensify/App/blob/eb5a569942af5f4117a548d9ed683046e11f9012/src/components/MoneyReportHeader.tsx#L126) comes after [my proposal](https://github.com/Expensify/App/issues/49169#issuecomment-2348789572). Prior to that it pointed to [shouldDisableApproveButton](https://github.com/Expensify/App/blob/eb5a569942af5f4117a548d9ed683046e11f9012/src/components/MoneyReportHeader.tsx#L124).

<details>
  <summary>Screenshot of time of the first propsal </summary>
  Image here
<img width="1072" alt="Screenshot 2024-09-13 at 3 16 46 in the afternoon" src="https://github.com/user-attachments/assets/0c3066ff-2d6e-4972-95ec-4775f50fde30">
  </details>
 


<details>
  <summary>Screenshot of time my proposal  </summary>

<img width="1072" alt="Screenshot 2024-09-13 at 3 17 27 in the afternoon" src="https://github.com/user-attachments/assets/53b36cec-ee26-4be4-9e68-8afe34a63758">
  </details>

**Comment by etCoderDysto** [comment]
Hi @abzokhattab. Writing few details takes time and make a difference in proposal submission time, and there seems to be a little difference between what you have suggested at first which is disabling the button ([we dont disable the submit button](https://github.com/Expensify/App/issues/49169#:~:text=we%20dont%20disable%20the%20submit%20button%20if%20the%20workspce%20is%20archived%20here) ), and what you suggested after the edit. That was why I raised the question. I have no ill intention towards you in saying that though🙏. 

Thanks!

**Comment by etCoderDysto** [comment]
> Hi @abzokhattab. Writing few details takes time and make a difference in proposal submission time, and there seems to be a little difference between what you have suggested at first which is disabling the button ([we dont disable the submit button](https://github.com/Expensify/App/issues/49169#:~:text=we%20dont%20disable%20the%20submit%20button%20if%20the%20workspce%20is%20archived%20here) ) by pointing a permalink to [shouldDisableApproveButton](https://github.com/Expensify/App/blob/eb5a569942af5f4117a548d9ed683046e11f9012/src/components/MoneyReportHeader.tsx#L124), and what you suggested after the edit. That was why I raised the question. I have no ill intention towards you in saying that though🙏. 
> 
> Thanks!
Before edit, it seems that you were suggesting disabling the button as I have commented here.

**Comment by trjExpensify** [comment]
Hm, wait. Why is the expense report being archived in the OP video, @srikarparsi? 🤔 We _should_ have stopped doing that, shouldn't we?

**Comment by trjExpensify** [comment]
As for how to handle this case of an `open` report on a deleted workspace, let's tread carefully. On OldDot I believe we automatically move it to the submitter's "new" primary policy or something like that. The report action buttons shown are then applicable to whichever workspace it was moved to. 

I think to handle this properly and holistically for all report states and setup scenarios, we're going to need better planning and probably the inclusion of the feature to let people manually move a report to a different workspace.

**Comment by srikarparsi** [comment]
> Why is the expense report being archived in the OP video, @srikarparsi? 🤔 We should have stopped doing that, shouldn't we?

Hmm, I think we wanted to stop `closing` the expense report when the workspace is deleted, but continue `archiving` it. If it wasn't archived, then transactions could be continued to be submitted on the report and people could continue to chat on the report (which is what I thought we didn't want?). What do you think?

> I think to handle this properly and holistically for all report states and setup scenarios, we're going to need better planning and probably the inclusion of the feature to let people manually move a report to a different workspace.

Yeah I think I agree

**Comment by trjExpensify** [comment]
Okay, I can see why we kept doing that for now because of otherwise having the ability to add expenses to the report, but we want to continue to allow people to chat on an expense report to discuss what to do about outstanding reports - and more importantly, we need to stop the actions of submit/approve/pay which all will fail - so I don't think we should archive them. Thinking through how we should handle this broadly and consistently: 

**When a workspace is deleted** 

1. All of `Submit`, `Approve` and `Pay` action buttons are hidden for all participants of the report, as those actions are no longer available on this now deleted workspace. 
2. The expense report is not archived to retain the ability to chat on it, but we remove the `Submit expense` & `Track expense` options from the expense report's create menu (i.e the menu where `Add attachment` etc lives). 
3. In the report's details page, we add an option row to `Move report`
    a. Submitter: The option row is visible on `open` and `processing` reports.
    b. Admin: The option row is visible on `open`, `processing`, `approved` and `closed` reports. (reimbursed reports can't be moved because they are a record of payment made from this company workspace).
   c. Everyone: You should not be able to move a report that has been previously exported to a _connected accounting system_, so the option row shouldn't show on exported reports. 
4. When `Move report` is pressed it opens a filtered participant selector:
   a. Submitter: all workspaces they have access to + their selfDM*(note on this below)
   b. Admins: all workspaces they are an admin of. If they choose a workspace that the submitter is not a member of, we invite the submitter to the workspace and do the following: 
   - Set that workspace owner as the member's approver if it's a workspace with "Advanced approvals", otherwise whoever is the approver set for "Submit & close" and "Submit & approve". 
   - Move the report to the `open` state if the workspace has `delayed submission` enabled. A `Submit` button is now available on the report. 
   - Move the report to the `processing` state if the workspace has delayed submission disabled (A.k.a "instant submit"). An `Approve` || `Pay` button is now visible depending on the workspace settings.
5. When the expense report is moved to somewhere else, we add a system message from the actor who moved the expense report to capture the action in the audit history that reads:
      - "moved this report to the $workspaceChatReportLink workspace" ([code](https://github.com/Expensify/Web-Expensify/blob/b9567b5a89527040437890183af674a2c7152cf4/lib/Report/Action/Moved.php#L38), it exists for the submit > pay with biz bank account flow)
      - "moved this report to the $selfDMChatReportLink" (doesn't exist yet)
    

*Q: If the submitter chooses their `selfDM` here in 4.a how should we handle it? All expenses in the selfDM are unreported as it stands I believe, so would we be effectively deleting the expense report to unreport the expenses and show them in the selfDM where they can then be submitted on to wherever they want to put them via the actionable whisper message? 

How does that sound? Let's align, and then I can create mocks for the relevant parts above. CC: @JmillsExpensify

**Comment by srikarparsi** [comment]
> The expense report is not archived to retain the ability to chat on it

Will reimbursed expenses not be archived as well? I feel like this distinction would be kind of weird, since you would no longer be able to chat on chat threads or task threads but would be able to continue chatting on expense threads after a workspace is deleted. I'm also not sure if allowing users to chat on these expense threads is useful since they have to be moved to a different report to be paid anyway (where they can also chat).

But I agree with everything else!

**Comment by trjExpensify** [comment]
Right, if the expense report isn't archived neither are the expenses on them. People retain the ability to comment on expense reports after a workspace is deleted today on OldDot because ultimately it can be moved to another workspace to progress to resolution, and for example, if the workspace was deleted and they still need to be processed to paid, the conversation can be had. So the suggestion is to keep that behaviour intact.

**Comment by johncschuster** [comment]
@Ollyws, can you review the above conversations and see if the previous proposals still match?

**Comment by johncschuster** [comment]
Thanks! Ok, it sounds like we're still looking for proposals then

**Comment by johncschuster** [comment]
Still looking for proposals. Increased the bounty

**Comment by johncschuster** [comment]
Still waiting on proposals

**Comment by johncschuster** [comment]
[Posted in Slack to drum up proposals](https://expensify.slack.com/archives/C01GTK53T8Q/p1728678586812809)

**Comment by srikarparsi** [comment]
Added an additional expected behavior section in the issue body to clarify what we're looking for on the frontend cc @trjExpensify :

> 1. When a workspace is deleted, no expense report should show `Submit`, `Approve` or `Pay` action buttons.
> 2. Additionally `Submit expense` & `Track expense` options should be hidden from the expense report’s create menu (i.e the menu where Add attachment etc lives).

**Comment by trjExpensify** [comment]
>Or are we also handling the scenarios mention from step 3 - 5 https://github.com/Expensify/App/issues/49169#issuecomment-2361066868

I think Srikar said 3-5 will take some backend work, so we don't have to do all of it in the same PR.

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
[$500] Expense - Submit button appears for archived workspace chat if delayed submission is enabled

### What is the root cause of that problem?
1. We are not checking whether the chat report is an archived one when displaying submit and approve button both in MoneyReportHeader and ReportPreview
https://github.com/Expensify/App/blob/d1f6190f6503fea55d16ca9e66ea7d70788bad23/src/components/ReportActionItem/ReportPreview.tsx#L328
https://github.com/Exensify/App/blob/d1f6190f6503fea55d16ca9e66ea7d70788bad23/src/components/ReportActionItem/ReportPreview.tsx#L176
https://github.com/Expensify/App/blob/d1f6190f6503fea55d16ca9e66ea7d70788bad23/src/components/MoneyReportHeader.tsx#L134
https://github.com/Expensify/App/blob/d1f6190f6503fea55d16ca9e66ea7d70788bad23/src/components/MoneyReportHeader.tsx#L138
Fortunately for pay button we already have the necessary check in IOU.canIOUBePaid so no change is needed
https://github.com/Expensify/App/blob/d1f6190f6503fea55d16ca9e66ea7d70788bad23/src/libs/actions/IOU.ts#L7052
2. We are not checking the archived state of the report and chatReport linked to it here [getMoneyRequestOptions](https://github.com/Expensify/App/blob/d1f6190f6503fea55d16ca9e66ea7d70788bad23/src/libs/ReportUtils.ts#L6895)
### What changes do you think we should make in order to solve the problem?

1. Create a var to check the status of the chatReport and use as a condition to display the buttons in MoneyReportHeader and ReportPreview
```
 const isArchivedChatReport = ReportUtils.isArchivedRoom(chatReport);
    const shouldShowSubmitButton = isOpenExpenseReport && reimbursableSpend !== 0 && !showRTERViolationMessage && !shouldShowBrokenConnectionViolation && !isArchivedChatReport;
      const shouldShowApproveButton = useMemo(() => !isArchivedChatReport && IOU.canApproveIOU(iouReport, policy), [iouReport, policy, isArchivedChatReport]);

```
2. In here
https://github.com/Expensify/App/blob/d1f6190f6503fea55d16ca9e66ea7d70788bad23/src/libs/ReportUtils.ts#L6895-L6998
```
const chatReport = getReport(report?.chatReportID ?? '-1');
    const isArchivedReport = isArchivedRoom(report) || isArchivedRoom(chatReport);
    if (isChatThread(report) || isTaskReport(report) || isInvoiceReport(report) || isSystemChat(report) || isArchivedReport) {
        return [];
    }
```
### What alternative solutions did you explore? (Optional)

**Comment by johncschuster** [comment]
@Ollyws can you take a look at the proposal above when you get a moment?

**Comment by FitseTLT** [comment]
> @johncschuster Given we're only targeting [these cases](https://github.com/Expensify/App/issues/49169#issuecomment-2418764175) I think we're good to continue with my [initial selection](https://github.com/Expensify/App/issues/49169#issuecomment-2358227756) of @etCoderDysto

@Ollyws @etCoderDysto 's proposal is not the right proposal
1. it is based on `isArchivedReport` but the money request report is not archived when the workspace is archived so we need to check whether the chat report is archived as I have suggested in my proposal
2. it is suggesting that we don't need to hide the submit and other menu because the compose is disabled but it is wrong the compose is only hidden for the workspace chat but for the money request report we still need to hide it 
so that's why I made my proposal. Please take a look at it. Thx!

**Comment by srikarparsi** [comment]
Asked a question [here](https://expensify.slack.com/archives/C03TQ48KC/p1729746202946249) and will proceed based on it

**Comment by srikarparsi** [comment]
Cool, lets move forward. Just to add on, we are no longer adding the `private_isArchived` rNVP to expense reports when a policy is deleted or a member is removed from a policy. We are only adding this `private_isArchived` rNVP to chat reports.

So, for expense reports, we should always look to see if the parent chat report is archived (has the `private_isArchived` rNVP, not the current expense report)

**Comment by srikarparsi** [comment]
Actually, @FitseTLT, can you please hold on the implementation. We're still discussing in this internal [thread](https://expensify.slack.com/archives/C03TQ48KC/p1729875348774879?thread_ts=1729746202.946249&cid=C03TQ48KC) about how we want to go about this so the implementation might change slightly.

**Comment by FitseTLT** [comment]
> Actually, @FitseTLT, can you please hold on the implementation. We're still discussing in this internal [thread](https://expensify.slack.com/archives/C03TQ48KC/p1729875348774879?thread_ts=1729746202.946249&cid=C03TQ48KC) about how we want to go about this so the implementation might change slightly.

Sure

**Comment by johncschuster** [comment]
Bumping this for Melvin while we discuss

**Comment by johncschuster** [comment]
@srikarparsi, it looks like the discussion in [this Slack thread](https://expensify.slack.com/archives/C03TQ48KC/p1729746202946249) has been resolved. Can you summarize the outcome of that thread and get us up to speed on who needs to do what here?

**Comment by srikarparsi** [comment]
Yup, I'm still waiting to merge this [PR](https://github.com/Expensify/Auth/pull/12960) but this takes back everything I said in this [comment](https://github.com/Expensify/App/issues/49169#issuecomment-2436391728). Expense reports will still have the `private_isArchived` rNVP. So basically, the isArchivedRoom check should make sure the type is not "expense" and we should create a new `isArchivedExpenseReport` check that hides the action buttons. Feel free to start on that while the Auth PR gets merged if you would like.

**Comment by johncschuster** [comment]
@FitseTLT bump on the comment above!

**Comment by FitseTLT** [comment]
Ready to make FE changes @srikarparsi Let us know when your BE PR is ready. Thx.

**Comment by srikarparsi** [comment]
Cool it's all ready, just needs to be reviewed and merged. Maybe 2-3 days for it to hit production but feel free to start on the PR. I can review it early on and we can test after the [backend PR](https://github.com/Expensify/Auth/pull/12960) hits production

**Comment by FitseTLT** [comment]
Okay @srikarparsi Draft [PR](https://github.com/Expensify/App/pull/52183) created.

**Comment by srikarparsi** [comment]
The frontend [PR](https://github.com/Expensify/App/pull/52183) is still being worked on but is close to finishing. I also have a backend [PR here](https://github.com/Expensify/Auth/pull/13337) which is in review.

**Comment by johncschuster** [comment]
Oh sweet! I'll keep an eye out for the merge. Thanks, @srikarparsi!

**Comment by johncschuster** [comment]
Looks like the backend PR was deployed to prod two days ago, and @FitseTLT is going back over the frontend PR now 🙌

**Comment by johncschuster** [comment]
There's active discussion and work happening on the linked frontend PR above. Not overdue 👍

**Comment by johncschuster** [comment]
@srikarparsi / @FitseTLT / @Ollyws can we get an update on [this PR](https://github.com/Expensify/App/pull/52183)?

**Comment by FitseTLT** [comment]
There is a pending BE [pr](https://github.com/Expensify/Auth/pull/13475). As soon as it is merged we will have a final test and record snapshots  👍

**Comment by johncschuster** [comment]
That's great! Thanks, @FitseTLT!

**Comment by johncschuster** [comment]
@FitseTLT and @Ollyws, I will be OOO starting December 23 and will be returning January 6th. A handful of folks on the BZ team will be online for a few days in between the 25th and the 1st, but we'll be operating with a skeleton crew. I will be issuing my payments when I return on January 6th. If the PR linked passes the regression threshold before I return and you would like the payment issued sooner, please post this issue in [#expensify-open-source](https://expensify.enterprise.slack.com/archives/C01GTK53T8Q) and someone on the team will jump in.

Thank you!

**Comment by johncschuster** [comment]
Hey everyone! I'm back from break and will continue to push this forward 👋

**Comment by FitseTLT** [comment]
> Ok! It looks like the BE PR from above was merged into Auth about two weeks ago. That means we're waiting on tests and snapshots, right @FitseTLT?

Yep correct on C+ review stage

**Comment by johncschuster** [comment]
Great! I'll leave it to you then. Thanks, @FitseTLT!

**Comment by johncschuster** [comment]
https://github.com/Expensify/App/pull/52183 has been deployed to staging 🥳

**Comment by johncschuster** [comment]
# BugZero Checklist:

- [ ] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [ ] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [ ] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [ ] 3z. Other:

</details>

- [ ] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment:

- [ ] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion:

- [ ] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

<details>
<summary>Regression Test Proposal Template</summary>


- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:


-

### Test:


1.

Do we agree 👍 or 👎


</details>

**Comment by johncschuster** [comment]
## Payment Summary

Contributor: @FitseTLT  paid $500 via [Upwork](https://www.upwork.com/nx/wm/offer/104580833)
Contributor+: @Ollyws  owed $500 via NewDot

**Comment by johncschuster** [comment]
Payment has been issued to @FitseTLT. Once the checklist is complete, please go ahead and request payment via ND, @Ollyws. Thank you!

**Comment by johncschuster** [comment]
Thanks, @Ollyws! We're good to close this up!

**Comment by garrettmknight** [comment]
$500 approved for @Ollyws
